src/org/rascalmpl/library/lang/rascal/tests/functionality/SyntaxRoleModifiers1.rsc

```diff
@@ -0,0 +1,90 @@
+module lang::rascal::tests::functionality::SyntaxRoleModifiers1
+
+import Type;
+
+lexical Z = [a-z]+;
+
+syntax A = "a" | "b" "b";
+
+layout L = [\ ]*;
+
+data A = a();
+
+test bool useOfDifferentModifierTypesWithinASingleScope() {
+    data[A] anExample = a();
+    syntax[A] anotherExample = [syntax[A]] "a";
```

Comment:
is it also legal to write `parse(#syntax[A], "a")` ? and how does this relate to `start[A]`

Reply:
Excellent question!

`syntax[X]` is equivalent to `X` in any context. So `#X` produces the same as `#syntax[X]` as long as `X` is not ambiguous. 

If X is ambiguous then it could happen that `#X` produces an ADT definition instead of a syntax definition.

For `X` you can substitute other non-terminal forms; like `syntax[start[X]]` and `syntax[A*]` but that is implemented on a different branch. In the current branch you can type `start[syntax[X]]` to get `start[X]` while `syntax[start[X]]` is undefined. Also `syntax[A*]` is undefined here.

On the other branch, after we have this one working, and merged into it, we can use `syntax[A*]` to introduce layout symbols in all the right places, and also to pattern match on lexical lists, mutually exclusively with context-free lists: i.e. `int size(syntax[&T*] l)` and `int size(lexical[&T*] l)` would be two different overloads to handle the different way of counting elements in either kind of list.

Reply:
ah interresting. I would have expected `syntax[start[A]]` to be valid indeed. good that it's incoming as well.